userspace/libsinsp/plugin.cpp

```diff
@@ -449,13 +453,35 @@ std::list<sinsp_plugin::info> sinsp_plugin::plugin_infos(sinsp* inspector)
 	return ret;
 }
 
-sinsp_plugin::sinsp_plugin()
-	: m_nfields(0)
+bool sinsp_plugin::is_plugin_loaded(sinsp* inspector, std::string &filepath)
+{
+#ifdef _WIN32
+	// This returns an HMODULE indeed, but they are the same thing
+	HINSTANCE handle = (HINSTANCE)GetModuleHandle(filepath.c_str());
+#else
+	void* handle = dlopen(filepath.c_str(), RTLD_LAZY | RTLD_NOLOAD);
+#endif
+	if (!handle)
+	{
+		return false;
+	}
+	for(auto p : inspector->get_plugins())
+	{
+		if (p->m_handle == handle) {
+			return true;
+		}
+	}
```

Comment:
Is this guaranteed to be true? It doesn't seem obvious that dlopen()ing a file twice will return the same object both times

Reply:
For dlopen at least:
> RTLD_NOLOAD (since glibc 2.2)
              Don't load the shared object.  This can be used to test if
              the object is already resident (dlopen() returns NULL if
              it is not, or the object's handle if it is resident).
              This flag can also be used to promote the flags on a
              shared object that is already loaded.  For example, a
              shared object that was previously loaded with RTLD_LOCAL
              can be reopened with RTLD_NOLOAD | RTLD_GLOBAL.

Btw i will just store filepath from create_plugins and use that instead.

Reply:
TIL :) now if we only verified the behavior on Windows, we'd be all good :)

Reply:
For windows, i found: 
> The GetModuleHandle function returns a handle to a mapped module without incrementing its reference count.